frontstage/models.py

```diff
@@ -170,6 +98,33 @@ def validate_password(form, field):
             raise ValidationError(app.config['PASSWORD_CRITERIA_ERROR_TEXT'])
 
 
+class SecureMessagingForm(FlaskForm):
+    save_draft = SubmitField(label='Save Draft')
+    send = SubmitField(label='Send')
+    subject = StringField('Subject')
+    body = TextAreaField('Message')
+    thread_message_id = HiddenField('Thread message id')
+    msg_id = HiddenField('Message id')
+    thread_id = HiddenField('Thread id')
+    hidden_subject = HiddenField('Hidden Subject')
+
+    @staticmethod
+    def validate_subject(form, field):
+        subject = form['hidden_subject'].data if form['hidden_subject'].data else field.data
+        if len(subject) > 96:
+            raise ValidationError('Subject field length must not be greater than 100')
+        if form.send.data and not subject:
+            raise ValidationError('Please enter a subject')
```

Comment:
I may be wrong but i think in the current functionality users can save a draft with no subject. Looks like this change will take that ability away. Need to add this validation only if it's a send submit.